pci/src/vfio.rs

```diff
@@ -1095,14 +1104,30 @@ impl VfioCommon {
     pub(crate) fn enable_msix(&self) -> Result<(), VfioPciError> {
         if let Some(msix) = &self.interrupt.msix {
             let mut irq_fds: Vec<EventFd> = Vec::new();
-            for i in 0..msix.bar.table_entries.len() {
+            let vector_num = msix.bar.vector_num;
+
+            debug!(
+                "enable msix msix_table entries {} enabled {}",
+                msix.bar.table_entries.len(),
+                vector_num
+            );
+
+            for i in 0..vector_num {
                 if let Some(eventfd) = msix.interrupt_source_group.notifier(i as InterruptIndex) {
                     irq_fds.push(eventfd);
                 } else {
                     return Err(VfioPciError::MissingNotifier);
                 }
             }
 
+            // Here we just get MSI-X enabled, but no vector enabled, by setting vector
+            // with an invalid fd to kernel.
+            // To avoid assert in OwnedFd, set -2 here rather than -1 as usual.
+            if 0 == vector_num {
```

Comment:
It would be better to make this code support having an empty `irq_fds` vector rather an abusing a sentinel value like this.

Reply:
Okay, the lastest OwnedFd still uses -1 as the invalid value,  I'll find some other methods to do this.